include/rcpputils/filesystem_helper.hpp

```diff
@@ -412,6 +412,35 @@ inline path temp_directory_path()
   return path(temp_path);
 }
 
+/**
+ * \brief Return current working directory.
+ *
+ * \return The current working directory.
+ *
+ * \throws std::system_error
+ */
+inline path current_path()
+{
+  char * cwd;
+#ifdef _WIN32
+#ifdef UNICODE
+#error "rcpputils::fs does not support Unicode paths"
+#endif
+  cwd = _getcwd(NULL, 0);
+#else
+  cwd = getcwd(NULL, 0);
```

Comment:
@cottsay nit: although all our currently supported platforms support this use of `getcwd()` (i.e. passing in a NULL buffer, yielding a dynamically allocated one), it's not part of POSIX. 

Creating a `PATH_MAX` (or `MAX_PATH` in Windows) long array for a buffer would keep it compliant.

Reply:
283a2f3